app/indexers/hyrax/valkyrie_indexer.rb

```diff
@@ -114,7 +114,7 @@ def indexer_class_for(resource)
 
         return indexer_class if indexer_class.is_a?(Class) &&
                                 indexer_class.instance_methods.include?(:to_solr)
-        ValkyrieIndexer
+        resource.is_a?(Hyrax::Work) ? Hyrax::ValkyrieWorkIndexer : ValkyrieIndexer
```

Comment:
do we prefer the type check or `resource.work?`

Reply:
Oooh, that sounds better.